src/backend/base/langflow/components/deactivated/ChatLiteLLMModel.py

```diff
@@ -137,18 +132,9 @@ def build_model(self) -> LanguageModel:
             "Cohere": "cohere_api_key",
             "OpenRouter": "openrouter_api_key",
         }
-
         # Set the API key based on the provider
-        api_keys: dict[str, Optional[str]] = {v: None for v in provider_map.values()}
-
-        if variable_name := provider_map.get(self.provider):
-            api_keys[variable_name] = self.api_key
-        else:
-            raise ChatLiteLLMException(
-                f"Provider {self.provider} is not supported. Supported providers are: {', '.join(provider_map.keys())}"
-            )
-
-        os.environ["OPENAI_API_KEY"] = self.api_key
+        api_keys: dict[str, Optional[str]] = {provider_map[self.provider]: self.api_key}
+        os.environ[self.provider.upper() + "_API_KEY"] = self.api_key
```

Comment:
We shouldn't change the user's environment.

Reply:
sorry about that, i fixed it on my last commit.